exporter/latency.go

```diff
@@ -30,9 +39,72 @@ func (e *Exporter) extractLatencyMetrics(ch chan<- prometheus.Metric, c redis.Co
 			var spikeLast, spikeDuration, max int64
 			if _, err := redis.Scan(latencyResult, &eventName, &spikeLast, &spikeDuration, &max); err == nil {
 				spikeDurationSeconds := float64(spikeDuration) / 1e3
-				e.registerConstMetricGauge(ch, "latency_spike_last", float64(spikeLast), eventName)
-				e.registerConstMetricGauge(ch, "latency_spike_duration_seconds", spikeDurationSeconds, eventName)
+				e.registerConstMetricGauge(outChan, "latency_spike_last", float64(spikeLast), eventName)
+				e.registerConstMetricGauge(outChan, "latency_spike_duration_seconds", spikeDurationSeconds, eventName)
 			}
 		}
 	}
 }
+
+func (e *Exporter) extractLatencyHistogramMetrics(outChan chan<- prometheus.Metric, infoAll string, redisConn redis.Conn) {
+	reply, err := redis.Values(doRedisCmd(redisConn, "LATENCY", "HISTOGRAM"))
+	if err != nil {
+		logHistogramErrOnce.Do(func() {
+			log.Errorf("WARNING, LOGGED ONCE ONLY: cmd LATENCY HISTOGRAM, err: %s", err)
+		})
+		log.Debugf("cmd LATENCY HISTOGRAM, err: %s", err)
+		return
+	}
+
+	for i := 0; i < len(reply); i += 2 {
+
+		cmd := string(reply[i].([]byte))
+		details, _ := redis.Values(reply[i+1], nil)
+
+		var totalCalls uint64
+		var bucketInfo []uint64
+
+		if _, err := redis.Scan(details, nil, &totalCalls, nil, &bucketInfo); err != nil {
+			break
+		}
+
+		buckets := map[float64]uint64{}
+
+		for j := 0; j < len(bucketInfo); j += 2 {
+			usec := float64(bucketInfo[j])
+			count := bucketInfo[j+1]
+			buckets[usec] = count
+		}
+
+		totalUsecs := extractTotalUsecForCommand(infoAll, cmd)
+
+		labelValues := []string{"cmd"}
+		e.registerConstHistogram(outChan, "latency_usec", labelValues, totalCalls, float64(totalUsecs), buckets, cmd)
+	}
+}
+
+func extractTotalUsecForCommand(infoAll string, cmd string) uint64 {
+	sanitizedCmd := strings.ReplaceAll(cmd, "|", "\\|")
```

Comment:
This looks like we're escaping the pipe symbol. Why do we need this? Can there ever be a pipe symbol in `cmd`? (I don't think so).
If we want a same cmd, maybe just do something like 
`cmd = regexp.MustCompile(`[^a-zA-Z0-9]+`).ReplaceAllString(cmd, "")` (and make the regex a const outside the function so it compiles exactly once)

Reply:
In the current setup, yes, it is needed. As written, the tests will fail if it's not present.

`cmd` will contain a pipe if it describes a sub-command. An example from my setup is:
* `latency|histogram`
* `latency|latest`

The main reason why it has to be escaped is that something like `config|get` will match with `get` which are two separate entries in the `LATENCY HISTOGRAM` output.

Reply:
Oh interesting, didn't know that. In that case this makes sense, leave it as is (could possible add a short comment).